tools/pre-swap-checker/contracts/PreSwapChecker.sol

```diff
@@ -295,20 +294,186 @@ contract PreSwapChecker {
       errorCount++;
     }
 
-    if (!isValid(order, domainSeparator)) {
-      errors[errorCount] = "SIGNATURE_INVALID";
+    return (errorCount, errors);
+  }
+
+  /**
+    * @notice If order is going through delegate via provideOrder
+    * ensure necessary checks are set
+    * @param order Types.Order
+    * @param delegate IDelegate
+    * @return uint256 errorCount if any
+    * @return bytes32[] memory array of error messages
+    */
+  function checkDelegate(
+    Types.Order memory order,
+    IDelegate delegate
+    ) public view returns (uint256, bytes32[] memory ) {
+
+    (uint256 errorCount, bytes32[] memory errors) = checkSwap(order);
+    (uint256 delegateErrorCount, bytes32[] memory delegateErrors) = coreDelegateChecks(order, delegate);
```

Comment:
Both the `errors` from `checkSwap` and `delegateErrors` from `coreDelegateChecks` are of a fixed size.  

At first glance it looks like it's possible that the `delegateErrors` from the `coreDelegateChecks` may not fit or overwrite data when writing to the `errors` variable if we miscount the number of errors up to that point.

Can I get clarification on:
`bytes32[] memory errors = new bytes32[](9);`
`bytes32[] memory errors = new bytes32[](20);`
and why those hard numbers were chosen? 

I checked, but just want to confirm.

Reply:
addressed and constants made from magic numbers